services/webhooks2tasks/src/handlers/resticbackupSnapshotFinished.js

```diff
@@ -99,18 +99,41 @@ async function resticbackupSnapshotFinished(webhook: WebhookRequestData) {
       'YYYY-MM-DD HH:mm:ss'
     );
 
-    // Determine source from the snapshot path.
-    // 1: `/data/*` --> means this is a PVC we use whatever is after `/data/` as the name of the backup
-    // 2: `*.tar` or `*.sql` --> current implementation with the name of the backup right before the ending, for example: foo.bar.tar (the backup is called `bar`)
+    // Determine source from the snapshot path. This is used in the UI to show
+    // the type of backup this is.
+    //
+    // 1: `/data/*` --> means this is a PVC we use whatever is after `/data/` as
+    // the name of the backup.
+    // 2: `*.tar` or `*.sql` --> current implementation with the name of the
+    // backup right before the ending. For example: foo.bar.tar (the backup is
+    // called `bar`). This logic is made slightly more complex by the fact that
+    // there was a bug in the k8up system that produced filenames like
+    //
+    // * drupal-example-site-master-solr-prebackuppod.solr.tar
+    // * drupal-example-site-master-solr-prebackuppod
+    // * drupal-example-site-master-mariadb-prebackuppod
+    // * drupal-example-site-master-mariadb-prebackuppod.mariadb.sql
+    //
+    // So we need to cater for these additional filenames.
     let source;
     const paths = R.prop('paths', snapshot);
-    const pattern = /^\/data\/([\w-]+)|([\w-]+).(?:sql|tar)$/;
+    const pattern = /^\/data\/([\w-]+)|([\w-]+)\.(?:sql|tar)$/;
```

Comment:
couldn't we just add `/([\w]+)-prebackuppod/` as a third regex inside of here? not sure we really need the additional if check, but I let @rocketeerbkw decide this

Reply:
2 was hard enough to wrap my brain around ;) We could have a super regex

```
^\/data\/([\w-]+)|([\w-]+)\.(?:sql|tar)$|([\w]+)-prebackuppod
```

Reply:
yes! that one sounds good, can you implement that one?

Reply:
Updated